examples/bluetooth/ble_advertising.py

```diff
@@ -46,3 +47,42 @@ def _append(adv_type, value):
     _append(_ADV_TYPE_APPEARANCE, struct.pack('<h', appearance))
 
     return payload
+
+
+def decode_field(payload, adv_type):
+    i = 0
+    result = []
+    while i + 1 < len(payload):
+        if payload[i + 1] == adv_type:
+            result.append(payload[i + 2:i + payload[i] + 1])
+        i += 1 + payload[i]
+    return result
+
+
+def decode_name(payload):
+    n = decode_field(payload, _ADV_TYPE_NAME)
+    if n:
+        return n[0].decode('utf-8')
```

Comment:
Note that bytes.decode() is only available with CPYTHON_COMPAT enabled.  The pattern that's used instead is `str(n[0], 'utf-8')` which I think would be better to use here.  Also maybe `return str(n[0], 'utf-8') if n else ''` to make it shorter?